clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java

```diff
@@ -121,8 +122,9 @@ public LeaderAndIsrRequest(Struct struct, short version) {
             List<Integer> replicas = new ArrayList<>(replicasArray.length);
             for (Object r : replicasArray)
                 replicas.add((Integer) r);
+            boolean isNew = partitionStateData.hasField(IS_NEW_KEY_NAME) ? partitionStateData.getBoolean(IS_NEW_KEY_NAME) : false;
```

Comment:
I am not sure about why it is safer to disable replica creation on a broker with failed directory. Does that mean a broker will not get assigned any new replica as long as it has one disk failed? Does that require controller awareness? But the controller does not know about that if it runs old IBP, right?